### ISSUE TITLE
IllegalStateException for intercepted inner class

### ISSUE THREAD

**Issue by timonzi** [quarkusio/quarkus#30789]
### Describe the bug

It is not possible to use `@Transactional` in an inner class.

Example:

```
@ApplicationScoped
public class MyClass implements MyInterface {

    private class MyInnerClass  {
        @Transactional
        private void doSomething() {
        }
    }

}

```

Snippet from the exception:

```
Caused by: java.lang.IllegalStateException: The INNER class org.acme.MyClass$MyInnerClass declares an interceptor binding but it must be ignored per CDI rules
```

I think it has to do with https://github.com/quarkusio/quarkus/pull/30141

In earlier versions (2.15.3.Final) it was possible and worked well from my point of view. So I don't understand why this is restricted now.

### Expected behavior

No error should occur.

### How to Reproduce?

Add a class as mentioned in the description and execute for example a QuarkusTest.

### Output of `uname -a` or `ver`

5.15.0-58-generic #64-Ubuntu SMP Thu Jan 5 11:43:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.17" 2022-10-18

### Quarkus version or git rev

2.16.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

**Comment by manovotn** [comment]
Hello @timonzi,

according to specification, inner classes are not valid candidates for managed beans.
See [this part](https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#what_classes_are_beans) of the specification text.
Static nested class would be fine and should work (previously and now).

However, I am not aware of us limiting functionality, this change should be just another validation (a user error prevention in this case) that was previously missing. In other words, in previous Quarkus version the `MyInnerClass` from your example wouldn't become a bean and wouldn't be intercepted. The app as such would start normally though.

If that wasn't the case, I'd be curious to see a reproducer.

**Comment by mkouba** [comment]
Matej is right. Inner classes must be ignored during bean discovery. In Quarkus however, we detect wrong usage of interceptor bindings (among others) and fail the build. If you really need to declare an interceptor binding on an inner class then you can use the following config property  `quarkus.arc.detect-wrong-annotations=false`.

**Comment by timonzi** [comment]
@manovotn I tested with our existing code base and ran into test failures when removing the _Transactional_ annotation. Because of this I thought it would be necessary, respectively possible. But unfortunately I removed also one _Transactional_ annotation of a called method, which is **not** inside the inner class.
So at the end I wasn't able to reproduce anything, which does not fit your description.

Thank you both for the clarification and sorry for wasting your time.
